### ISSUE TITLE
Sampling frequency

### ISSUE THREAD

**Issue by eioe** [Omnifinity/OpenVR-Tracking-Example#1]
Thanks for your helpful contribution!
We're writing some of the output to a file, in order to analyze it later. By doing so, it looks like the sampling frequency is around 10Hz. Are we right? If so, is it possible to increase this easily? 
I'm only starting to work with OpenVR/C++ - so bear with me if this is an uninformed question. 
Thanks a bunch!

**Comment by peroht** [comment]
Thank you for your comment. 
Have you updated to the latest source? 
If not you might not have notice that there is a boolean flag to enable continous access of tracking data without waiting for events (such as button presses)?
It seems like you are experiencing this issue?
/Peter

-------- Originalmeddelande --------
Från: eioe <notifications@github.com> 
Datum: 2017-03-22  13:21  (GMT+01:00) 
Till: Omnifinity/OpenVR-Tracking-Example <OpenVR-Tracking-Example@noreply.github.com> 
Kopia: Subscribed <subscribed@noreply.github.com> 
Rubrik: [Omnifinity/OpenVR-Tracking-Example] Sampling frequency (#1) 

Thanks for your helpful contribution!

We're writing some of the output to a file, in order to analyze it later. By doing so, it looks like the sampling frequency is around 10Hz. Are we right? If so, is it possible to increase this easily?

I'm only starting to work with OpenVR/C++ - so bear with me if this is an uninformed question.

Thanks a bunch!

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/Omnifinity/OpenVR-Tracking-Example","title":"Omnifinity/OpenVR-Tracking-Example","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/Omnifinity/OpenVR-Tracking-Example"}},"updates":{"snippets":[{"icon":"DESCRIPTION","message":"Sampling frequency (#1)"}],"action":{"name":"View Issue","url":"https://github.com/Omnifinity/OpenVR-Tracking-Example/issues/1"}}}

**Comment by eioe** [comment]
Thank you for your helpful reply! 
Yes, we had an updated source and the WaitingForVREvents-flag was set to false but we still got a low sampling rate. However, I just found the culprit a few lines below. There's a Sleep command that was set to 100ms in order "not to overheat your computer". When I lowered this to 10ms, the sampling frequency went up accordingly to ~100Hz. Perfect! I was looking for something like that in the LighthouseTracking.cpp - but your comment put me on the right path. Thanks a lot! It's a really helpful tool!
